ctrlf.el

```diff
@@ -1015,6 +1067,61 @@ search, change back to regexp search."
       (isearch-backward-regexp)
     (ctrlf-backward 'regexp)))
 
+;;;###autoload
+(defun ctrlf-forward-symbol ()
+  "Search forward for symbol.
+If already in a search, go to next candidate, or if no input then
+insert the previous search string. If in a non-symbol search,
+change back to symbol search."
+  (interactive)
+  (if (and (window-minibuffer-p) (not ctrlf--active-p))
+      (isearch-forward-symbol)
+    (ctrlf-forward 'symbol)))
+
+;;;###autoload
+(defun ctrlf-forward-symbol-at-point ()
+  "Search forward for symbol at point.
+If already in a search, replace the current input and change to a
+symbol search, otherwise start the search. If no symbol is found,
+display an error message and do not search."
+  (interactive)
+  (let ((arg nil)
+        (pos nil)
+        (skip-search nil))
+    (with-current-buffer (window-buffer (minibuffer-selected-window))
+      (setq pos (point)) ; Save original point.
+      ;; NOTE: This idea is borrowed from `isearch' but I don't see
+      ;; how else it could be implemented.
+      (let ((start (car (find-tag-default-bounds)))) ; Find start of symbol.
+        (when (and start (< start (point))) ; `start' can be nil.
+          (setq ctrlf--current-starting-point start) ; For `ctrlf-forward'.
```

Comment:
Hmmm... I was going to say we should avoid using internal functions in these wrappers, but then I looked at the other ones and they do that too. I guess at some point I forgot that I was trying to do that. Room for future improvement!

Reply:
Oh yeah, always room for improvement 😂